code/modules/reagents/oldchem/reagents/reagents_admin.dm

```diff
@@ -51,4 +52,24 @@
 /datum/reagent/adminordrazine/nanites
 	name = "Nanites"
 	id = "nanites"
-	description = "Nanomachines that aid in rapid cellular regeneration."
\ No newline at end of file
+	description = "Nanomachines that aid in rapid cellular regeneration."
+
+
+// For random item spawning. Takes a list of paths, and returns the same list without anything that contains admin only reagents
+
+/proc/adminReagentCheck(var/list/incoming)
+	var/list/outgoing[0]
+	for(var/tocheck in incoming)
+		if(ispath(tocheck))
+			var/check = new tocheck
+			if (istype(check, /atom))
+				var/atom/reagentCheck = check
+				var/datum/reagents/reagents = reagentCheck.reagents
+				var/admin = 0
+				for(var/datum/reagent/reagent in reagents.reagent_list)
```

Comment:
@Fox-McCloud 
Doesn't need it. If you look up the for list proc in the dm whitepaper, you'll find that unless you use as to specify a type (eg `for(var/reagent as /datum/reagent)`) it uses the declared type t skip over any entries in the list that don't fit the variable's type. As such, only /datum/reagents can enter the loop, so you don't need the istype